solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -297,7 +296,7 @@ public int getZNodeVersion(){
     return znodeVersion;
   }
   public int getChildNodesVersion() {
-    return perReplicaStates == null ? -1 : perReplicaStates.cversion;
```

Comment:
Zk returns 0 for non-existent children. And we compare that to fetch the children from zk. Thus it should be 0, otherwise we may ned up fetching each time

Reply:
Cool makes sense then! 👍🏼

Reply:
Doesn't changing that to 0 instead of -1 create a risk that this `if (getChildNodesVersion() >= newPerReplicaStates.cversion) return this;` evaluates to true when it shouldn't like if current state is null. I assumed the reason it return -1 for null was because newPerReplicaStates.cversion can never be -1, but it could be 0.

Reply:
'copyWith' method assumes old and new states, and I think we should keep the state which has higher version. As in different code path, we check higher zk version to fetch the latest prs state. It should be consistent in both path. 
Let me know if I'm missing something here.

Reply:
>Just curious of why we want to use 0 here instead of -1

PRS version starts with 0. `-1`  means PRS is absent (`null`)

Reply:
Just a quick side note here - tested locally on freshly created PRS collection, zk returned `5` for `stat.getCversion()`, while for non PRS collection `stat.getCversion` is `0`. Therefore using `0` in `getChildNodesVersion` for non PRS collection should work as intended 😊

Reply:
@hiteshk25 Related to Justin's comment, if `getChildNodesVersion()` returns 0 for non-existent PRS, and a collection is enabled for PRS, wont the first versions be 0 and therefore ignored? We saw a case of this in solrman when a shard went down during conversion and wasn't detected because the first new PRS state was being ignored.

Reply:
That version was related to prs state 'replica1, versionX, active, leader', this is zk version of state.json children.

Reply:
```
(CONNECTED [localhost:2181]) /solr/collections/CF9/state.json> stat core_node8:19:A:L
Stat(
  czxid=0x1b00cb0e39
  mzxid=0x1b00cb0e39
  ctime=1652827907162
  mtime=1652827907162
  version=0
  cversion=0
  aversion=0
  ephemeralOwner=0x0
  dataLength=0
  numChildren=0
  pzxid=0x1b00cb0e39
)
(CONNECTED [localhost:2181]) /solr/collections/CF9/state.json> exit
michaelpieper@zookeeper-c09-1.stage:~$
```

Reply:
Ah gotcha!